### ISSUE TITLE
[Bug]: app.toml file with literal string for osmosis-mempool.arbitrage-min-gas-fee gets broken on startup

### ISSUE THREAD

**Issue by sigv** [osmosis-labs/osmosis#8055]
### What happened?

The default app.toml file is generated with double-quotes for strings. However, we have tooling that rewrites the TOML file. We have the strings as single-quote literal.

This presents an issue with mempool configuration in app.toml. We have configuration:

```toml
[osmosis-mempool]
arbitrage-min-gas-fee = '.005'
```

When we start running the chain, this is overwritten to:

```toml
[osmosis-mempool]
0.1
arbitrage-min-gas-fee = '.005'
```

As result, next time Osmosis is restarted, app.toml file cannot be parsed and we encounter a failure.

### Osmosis Version

24.0.0

### How to reproduce?

Replace all double-quotes with single-quotes for app.toml, and start Osmosis chain binary:

```toml
[osmosis-mempool]
arbitrage-min-gas-fee = '.005'
```

**Comment by sigv** [comment]
Mitigation is possible by applying `--reject-config-defaults` flag to binary.

**Comment by sigv** [comment]
Resolved via #8118.
